Use OpenSSL 3 certificate APIs only with OpenSSL, preserving LibreSSL support.

Index: libi2pd/Reseed.cpp
--- libi2pd/Reseed.cpp.orig
+++ libi2pd/Reseed.cpp
@@ -16,7 +16,7 @@
 #include <boost/algorithm/string.hpp>
 #include <openssl/ssl.h>
 #include <openssl/err.h>
-#if (OPENSSL_VERSION_NUMBER >= 0x030000000) // since 3.0.0
+#if !defined(LIBRESSL_VERSION_NUMBER) && OPENSSL_VERSION_NUMBER >= 0x30000000L // since 3.0.0
 #include <openssl/core_names.h>
 #endif
 #include <zlib.h>
@@ -537,7 +537,7 @@ namespace data
 				// extract RSA key (we need n only, e = 65537)
 				EVP_PKEY * pubKey = X509_get_pubkey (cert);
 				const BIGNUM * n = nullptr;
-#if (OPENSSL_VERSION_NUMBER >= 0x030000000) // since 3.0.0
+#if !defined(LIBRESSL_VERSION_NUMBER) && OPENSSL_VERSION_NUMBER >= 0x30000000L // since 3.0.0
 				BIGNUM * n1 = BN_new ();
 				if (EVP_PKEY_get_bn_param (pubKey, OSSL_PKEY_PARAM_RSA_N, &n1) > 0)
 					n = n1;
@@ -557,7 +557,7 @@ namespace data
 				}
 				else
 					LogPrint (eLogError, "Reseed: Can't extract RSA key from ", filename);
-#if (OPENSSL_VERSION_NUMBER >= 0x030000000) // since 3.0.0
+#if !defined(LIBRESSL_VERSION_NUMBER) && OPENSSL_VERSION_NUMBER >= 0x30000000L // since 3.0.0
 				BN_free (n1);
 #endif
 			}
